src/deadline/cinema4d_submitter/font_installer.py

```diff
@@ -0,0 +1,286 @@
+            logger.debug(f"Couldn't recursively find tempFonts in subfolder: {subfolder}")
+            continue
+
+        for file_name in os.listdir(full_sub_dir):
+            full_assetpath = os.path.join(full_sub_dir, file_name)
+            _, ext = os.path.splitext(full_assetpath)
+            if ext.lower() in FONT_EXTENSIONS:
+                logger.debug(f"Adding: {full_assetpath}")
+                fonts.add(full_assetpath)
+            else:
+                logger.warning(
+                    f"A file that is not a supported font was found in the tempFonts folder: {full_assetpath}"
+                )
+    return fonts
+
+
+def get_font_name(dst_path: str) -> str:
+    """
+    Get a font's Windows system name, which is the name stored in the registry.
+
+    :param dst_path: path of font that needs to be named
+
+    :returns: string with the font's name
+    """
+    if sys.platform != "win32":
+        raise RuntimeError("Font installation is only supported on Windows")
+
+    try:
+        filename = os.path.basename(dst_path)
+        fontname = os.path.splitext(filename)[0]
+
+        # Try to get the font's real name
+        cb = wintypes.DWORD()
+        if gdi32.GetFontResourceInfoW(filename, ctypes.byref(cb), None, GFRI_DESCRIPTION):
+            buf = (ctypes.c_wchar * cb.value)()
+            if gdi32.GetFontResourceInfoW(filename, ctypes.byref(cb), buf, GFRI_DESCRIPTION):
+                fontname = buf.value
+        is_truetype = wintypes.BOOL()
+        cb.value = ctypes.sizeof(is_truetype)
+        gdi32.GetFontResourceInfoW(
+            filename, ctypes.byref(cb), ctypes.byref(is_truetype), GFRI_ISTRUETYPE
+        )
+        if is_truetype:
+            fontname += " (TrueType)"
+
+    except Exception as e:
+        raise e
+
+    return fontname
+
+
+def install_font(src_path: str, scope: str = INSTALL_SCOPE_USER) -> Tuple[bool, str]:
+    """
+    Install provided font to the worker machine
+
+    :param src_path: path of font that needs to be installed
+
+    :returns: boolean that represents if the font was installed and a string with any traceback that was created
+    """
+    if sys.platform != "win32":
+        return False, "Font installation is only supported on Windows"
+
+    try:
+        # Determine font destination
+        if scope == INSTALL_SCOPE_SYSTEM:
+            dst_dir = FONT_LOCATION_SYSTEM
+            registry_scope = winreg.HKEY_LOCAL_MACHINE
+        else:
+            # Check if the Fonts folder exists, create it if it doesn't
+            if not os.path.exists(FONT_LOCATION_USER):
+                logger.info(f"Creating User Fonts folder: {FONT_LOCATION_USER}")
+                os.makedirs(FONT_LOCATION_USER)
+
+            dst_dir = FONT_LOCATION_USER
+            registry_scope = winreg.HKEY_CURRENT_USER
+        dst_path = os.path.join(dst_dir, os.path.basename(src_path))
+
+        # Copy the font to the Windows Fonts folder
+        shutil.copy(src_path, dst_path)
```

Comment:
Is there a way to use font locally without installing them on the worker? I see we clean up the fonts afterwards which is probably ok. But even better if we don't have to mess with the underlying worker at all.